### ISSUE TITLE
camera_qcom.cc clEnqueueWriteBuffer fails sometimes

### ISSUE THREAD

**Issue by pd0wm** [commaai/openpilot#19724]
Seeing quite a bit of these pop up in sentry. No uptick in bug reports about camerad crashing, so might only happen on startup/shutdown.

Tombstone:
```
CM Version: '13.0-20200831-UNOFFICIAL-Sultan-oneplus3'
Build fingerprint: 'OnePlus/OnePlus3/OnePlus3:6.0.1/MMB29M/362280:user/release-keys'
Revision: '0'
ABI: 'arm64'
pid: 6914, tid: 6968, name: processing  >>> ./camerad <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
Abort message: 'selfdrive/camerad/cameras/camera_qcom.cc:2065: void update_lapmap(MultiCameraState *, const CameraBuf *, const int): assertion "CL_SUCCESS == clEnqueueWriteBuffer(b->q, s->rgb_conv_roi_cl, true, 0, width * height * 3 * sizeof(uint8_t), rgb_roi_buf.get(), 0, 0, 0)" failed'
    x0   0000000000000000  x1   0000000000001b38  x2   0000000000000006  x3   0000000000000000
    x4   0000000000000000  x5   0000000000000001  x6   0000000000000000  x7   0000000000000000
    x8   0000000000000083  x9   21282f1f2b2f1f2b  x10  7f7f7f7f7f7f7f7f  x11  0101010101010101
    x12  0000007f81da5888  x13  7c37eaf726574e1d  x14  7c37eaf726574e1d  x15  00056836e819dabb
    x16  0000007f81d9f560  x17  0000007f81d3116c  x18  0000007f73080d3f  x19  0000007f73081500
    x20  0000007f73081440  x21  0000000000000003  x22  0000000000000006  x23  0000000000000091
    x24  0000007fda87e5f8  x25  0000005562bd7000  x26  0000000000005221  x27  000000000000f663
    x28  0000000000000091  x29  0000007f73080f80  x30  0000007f81d2e908
    sp   0000007f73080f80  pc   0000007f81d31174  pstate 0000000020000000
    v0   28666f657a6973202a2033202a207468  v1   6f725f626772202c29745f38746e6975
    v2   2c65757274202c6c635f696f725f766e  v3   67696568202a206874646977202c3020
    v4   80200802802008000000000000000000  v5   00000000000000000000040000000000
    v6   00000000000000000000000000000400  v7   202c30202c29287465672e6675625f69
    v8   00000000000000000000000000000000  v9   00000000000000000000000000000000
    v10  00000000000000000000000000000000  v11  00000000000000000000000000000000
    v12  00000000000000000000000000000000  v13  00000000000000000000000000000000
    v14  00000000000000000000000000000000  v15  00000000000000000000000000000000
    v16  40100401401004014010040140100401  v17  a00aa00800000004aaa8000400000400
    v18  80200802802008000000000000000400  v19  00000000000000000000000000000000
    v20  00000000000000000000000000000000  v21  ffffffffffffffffffffffffffffffff
    v22  80402010080402018040201008040201  v23  80402010080402018040201008040201
    v24  0000000000000000000000000000fffe  v25  00000000000000000000000000000000
    v26  00000000000000000000000000000000  v27  00000000000000000000000000000000
    v28  00000000000000000000000000000000  v29  00000000000000000000000000000000
    v30  00000000000000000000000000000000  v31  00000000000000000000000000000000
    fpsr 00000010  fpcr 00000000

backtrace:
    #00 pc 000000000006a174  /system/lib64/libc.so (tgkill+8)
    #01 pc 0000000000067904  /system/lib64/libc.so (pthread_kill+68)
    #02 pc 0000000000023848  /system/lib64/libc.so (raise+28)
    #03 pc 000000000001dfe8  /system/lib64/libc.so (abort+60)
    #04 pc 0000000000021430  /system/lib64/libc.so (__libc_fatal+128)
    #05 pc 000000000001e0c0  /system/lib64/libc.so (__assert2+40)
    #06 pc 00000000000116bc  /data/openpilot/selfdrive/camerad/camerad (_Z20camera_process_frameP16MultiCameraStateP11CameraStatei+2396)
    #07 pc 000000000000c5e0  /data/openpilot/selfdrive/camerad/camerad (_Z17processing_threadP16MultiCameraStatePKcP11CameraStatePFvS0_S4_iE+96)
    #08 pc 000000000000d004  /data/openpilot/selfdrive/camerad/camerad (_ZNSt6__ndk114__thread_proxyINS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEPFPvP16MultiCameraStatePKcP11CameraStatePFvS9_SD_iEES9_SB_SD_SF_EEEEES7_S7_+52)
    #09 pc 0000000000066d54  /system/lib64/libc.so (_ZL15__pthread_startPv+52)
    #10 pc 000000000001eb94  /system/lib64/libc.so (__start_thread+16)

stack:
         0000007f73080f00  0000000000000000
         0000007f73080f08  0000000000000000
         0000007f73080f10  0000007f73080f60  [stack:6968]
         0000007f73080f18  0000007f81d2e518  /system/lib64/libc.so (_Z23__pthread_internal_findl+80)
         0000007f73080f20  0000007f73081440  [stack:6968]
         0000007f73080f28  0000007f81da7000
         0000007f73080f30  0000000000000003
         0000007f73080f38  0000000000000000
         0000007f73080f40  0000007f73080f60  [stack:6968]
         0000007f73080f48  0000007f81d2e4e4  /system/lib64/libc.so (_Z23__pthread_internal_findl+28)
         0000007f73080f50  0000000000000000
         0000007f73080f58  0000000000000000
         0000007f73080f60  0000007f73080f80  [stack:6968]
         0000007f73080f68  0000007f81d2e8ec  /system/lib64/libc.so (pthread_kill+44)
         0000007f73080f70  0000007f73081500  [stack:6968]
         0000007f73080f78  0000007f73081440  [stack:6968]
    #00  0000007f73080f80  0000007f73080fb0  [stack:6968]
         ........  ........
    #01  0000007f73080f80  0000007f73080fb0  [stack:6968]
         0000007f73080f88  0000007f81cea84c  /system/lib64/libc.so (raise+32)
         0000007f73080f90  0000007f73081018  [stack:6968]
         0000007f73080f98  0000007f73081020  [stack:6968]
         0000007f73080fa0  0000007fda87e5e8  [stack]
         0000007f73080fa8  0000000000000091
    #02  0000007f73080fb0  0000007f73080fe0  [stack:6968]
         0000007f73080fb8  0000007f81ce4fec  /system/lib64/libc.so (abort+64)
         0000007f73080fc0  0000007f73081018  [stack:6968]
         0000007f73080fc8  0000007f81870f3c  /system/vendor/lib64/libCB.so (cb_enqueue_write_buffer+284)
         0000007f73080fd0  ffffffffffffffdf
         0000007f73080fd8  0000000000000006
    #03  0000007f73080fe0  0000007f73081040  [stack:6968]
         0000007f73080fe8  0000007f81ce8434  /system/lib64/libc.so (__fortify_chk_fail)
         0000007f73080ff0  0000007fda87e838  [stack]
         0000007f73080ff8  0000007f7d3e067f  anon_inode:dmabuf
         0000007f73081000  0000007f78a096e0  [anon:libc_malloc]
         0000007f73081008  0000000000000001
         0000007f73081010  0000007f78996200  [anon:libc_malloc]
         0000007f73081018  ffffffffffffffdf
         0000007f73081020  0000000000000091
         0000007f73081028  0000007fda87e5f8  [stack]
         0000007f73081030  0000005562bd7000  /data/openpilot/selfdrive/camerad/camerad
         0000007f73081038  7c37eaf726574e1d
    #04  0000007f73081040  0000007f73081150  [stack:6968]
         0000007f73081048  0000007f81ce50c4  /system/lib64/libc.so (longjmperror)
         0000007f73081050  0000007f73081150  [stack:6968]
         0000007f73081058  0000007f73081150  [stack:6968]
         0000007f73081060  0000007f73081110  [stack:6968]
         0000007f73081068  ffffff80ffffffc8
         0000007f73081070  0000007f73081150  [stack:6968]
         0000007f73081078  0000007f73081150  [stack:6968]
         0000007f73081080  0000007f73081110  [stack:6968]
         0000007f73081088  ffffff80ffffffc8
         0000007f73081090  0000000000000000
         0000007f73081098  0000000000000000
         0000007f730810a0  72702e6775626564
         0000007f730810a8  6f697461746f7265
         0000007f730810b0  6c62617369642e6e
         0000007f730810b8  0000000000000065
         ........  ........
    #05  0000007f73081150  0000007f73081380  [stack:6968]
         0000007f73081158  0000005562b996c0  /data/openpilot/selfdrive/camerad/camerad (_Z20camera_process_frameP16MultiCameraStateP11CameraStatei+2400)
    #06  0000007f73081160  0000000000000000
         0000007f73081168  0000007f81d2d000  /system/lib64/libc.so (strpbrk+32)
         0000007f73081170  0000000000000154
         0000007f73081178  0000000000000004
         0000007f73081180  0000000000000000
         0000007f73081188  0000007f73285a00  [anon:libc_malloc]
         0000007f73081190  0000007fda87eef8  [stack]
         0000007f73081198  0000007fda87f7d8  [stack]
         0000007f730811a0  0000007f7edd1700  [anon:libc_malloc]
         0000007f730811a8  0000000000000000
         0000007f730811b0  0000005562bd766c  /data/openpilot/selfdrive/camerad/camerad
         0000007f730811b8  0000007f81d2d000  /system/lib64/libc.so (strpbrk+32)
         0000007f730811c0  0000007fda87e4a0  [stack]
         0000007f730811c8  0000007fda87f7d8  [stack]
         0000007f730811d0  0000007f730811e0  [stack:6968]
         0000007f730811d8  0000007f8186e6dc  /system/vendor/lib64/libCB.so (cl_arraylist_destroy+28)
         ........  ........
    #07  0000007f73081390  0000007f78957150  [anon:libc_malloc]
         0000007f73081398  0000000000000000
         0000007f730813a0  000000000000000b
         0000007f730813a8  00000000000ff000
         0000007f730813b0  0000007f81e770d8
         0000007f730813b8  0000007f81d9f000  /system/lib64/libc.so
         0000007f730813c0  0000007f730814c0  [stack:6968]
         0000007f730813c8  0000007f78957150  [anon:libc_malloc]
         0000007f730813d0  0000007f73081400  [stack:6968]
         0000007f730813d8  0000005562b95008  /data/openpilot/selfdrive/camerad/camerad (_ZNSt6__ndk114__thread_proxyINS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEPFPvP16MultiCameraStatePKcP11CameraStatePFvS9_SD_iEES9_SB_SD_SF_EEEEES7_S7_+56)
    #08  0000007f730813e0  0000007f81d9f000  /system/lib64/libc.so
         0000007f730813e8  0000007f730814c0  [stack:6968]
         0000007f730813f0  0000007f730814c0  [stack:6968]
         0000007f730813f8  0000007f73081440  [stack:6968]
         0000007f73081400  0000007f73081410  [stack:6968]
         0000007f73081408  0000007f81d2dd58  /system/lib64/libc.so (_ZL15__pthread_startPv+56)
    #09  0000007f73081410  0000007f73081430  [stack:6968]
         0000007f73081418  0000007f81ce5b98  /system/lib64/libc.so (__start_thread+20)
         0000007f73081420  0000007f81e77018
         0000007f73081428  0000000000001b02
    #10  0000007f73081430  0000000000000000
         0000007f73081438  0000000000000000
         0000007f73081440  0000007f7327f440
         0000007f73081448  0000007f72f82440
         0000007f'CM Version: '13.0-20200831-UNOFFICIAL-Sultan-oneplus3'
Build fingerprint: 'OnePlus/OnePlus3/OnePlus3:6.0.1/MMB29M/362280:user/release-keys'
Revision: '0'
ABI: 'arm64'
pid: 6914, tid: 6968, name: processing  >>> ./camerad <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
Abort message: 'selfdrive/camerad/cameras/camera_qcom.cc:2065: void update_lapmap(MultiCameraState *, const CameraBuf *, const int): assertion "CL_SUCCESS == clEnqueueWriteBuffer(b->q, s->rgb_conv_roi_cl, true, 0, width * height * 3 * sizeof(uint8_t), rgb_roi_buf.get(), 0, 0, 0)" failed'
    x0   0000000000000000  x1   0000000000001b38  x2   0000000000000006  x3   0000000000000000
    x4   0000000000000000  x5   0000000000000001  x6   0000000000000000  x7   0000000000000000
    x8   0000000000000083  x9   21282f1f2b2f1f2b  x10  7f7f7f7f7f7f7f7f  x11  0101010101010101
    x12  0000007f81da5888  x13  7c37eaf726574e1d  x14  7c37eaf726574e1d  x15  00056836e819dabb
    x16  0000007f81d9f560  x17  0000007f81d3116c  x18  0000007f73080d3f  x19  0000007f73081500
    x20  0000007f73081440  x21  0000000000000003  x22  0000000000000006  x23  0000000000000091
    x24  0000007fda87e5f8  x25  0000005562bd7000  x26  0000000000005221  x27  000000000000f663
    x28  0000000000000091  x29  0000007f73080f80  x30  0000007f81d2e908
    sp   0000007f73080f80  pc   0000007f81d31174  pstate 0000000020000000
    v0   28666f657a6973202a2033202a207468  v1   6f725f626772202c29745f38746e6975
    v2   2c65757274202c6c635f696f725f766e  v3   67696568202a206874646977202c3020
    v4   80200802802008000000000000000000  v5   00000000000000000000040000000000
    v6   00000000000000000000000000000400  v7   202c30202c29287465672e6675625f69
    v8   00000000000000000000000000000000  v9   00000000000000000000000000000000
    v10  00000000000000000000000000000000  v11  00000000000000000000000000000000
    v12  00000000000000000000000000000000  v13  00000000000000000000000000000000
    v14  00000000000000000000000000000000  v15  00000000000000000000000000000000
    v16  40100401401004014010040140100401  v17  a00aa00800000004aaa8000400000400
    v18  80200802802008000000000000000400  v19  00000000000000000000000000000000
    v20  00000000000000000000000000000000  v21  ffffffffffffffffffffffffffffffff
    v22  80402010080402018040201008040201  v23  80402010080402018040201008040201
    v24  0000000000000000000000000000fffe  v25  00000000000000000000000000000000
    v26  00000000000000000000000000000000  v27  00000000000000000000000000000000
    v28  00000000000000000000000000000000  v29  00000000000000000000000000000000
    v30  00000000000000000000000000000000  v31  00000000000000000000000000000000
    fpsr 00000010  fpcr 00000000

backtrace:
    #00 pc 000000000006a174  /system/lib64/libc.so (tgkill+8)
    #01 pc 0000000000067904  /system/lib64/libc.so (pthread_kill+68)
    #02 pc 0000000000023848  /system/lib64/libc.so (raise+28)
    #03 pc 000000000001dfe8  /system/lib64/libc.so (abort+60)
    #04 pc 0000000000021430  /system/lib64/libc.so (__libc_fatal+128)
    #05 pc 000000000001e0c0  /system/lib64/libc.so (__assert2+40)
    #06 pc 00000000000116bc  /data/openpilot/selfdrive/camerad/camerad (_Z20camera_process_frameP16MultiCameraStateP11CameraStatei+2396)
    #07 pc 000000000000c5e0  /data/openpilot/selfdrive/camerad/camerad (_Z17processing_threadP16MultiCameraStatePKcP11CameraStatePFvS0_S4_iE+96)
    #08 pc 000000000000d004  /data/openpilot/selfdrive/camerad/camerad (_ZNSt6__ndk114__thread_proxyINS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEPFPvP16MultiCameraStatePKcP11CameraStatePFvS9_SD_iEES9_SB_SD_SF_EEEEES7_S7_+52)
    #09 pc 0000000000066d54  /system/lib64/libc.so (_ZL15__pthread_startPv+52)
    #10 pc 000000000001eb94  /system/lib64/libc.so (__start_thread+16)

stack:
         0000007f73080f00  0000000000000000
         0000007f73080f08  0000000000000000
         0000007f73080f10  0000007f73080f60  [stack:6968]
         0000007f73080f18  0000007f81d2e518  /system/lib64/libc.so (_Z23__pthread_internal_findl+80)
         0000007f73080f20  0000007f73081440  [stack:6968]
         0000007f73080f28  0000007f81da7000
         0000007f73080f30  0000000000000003
         0000007f73080f38  0000000000000000
         0000007f73080f40  0000007f73080f60  [stack:6968]
         0000007f73080f48  0000007f81d2e4e4  /system/lib64/libc.so (_Z23__pthread_internal_findl+28)
         0000007f73080f50  0000000000000000
         0000007f73080f58  0000000000000000
         0000007f73080f60  0000007f73080f80  [stack:6968]
         0000007f73080f68  0000007f81d2e8ec  /system/lib64/libc.so (pthread_kill+44)
         0000007f73080f70  0000007f73081500  [stack:6968]
         0000007f73080f78  0000007f73081440  [stack:6968]
    #00  0000007f73080f80  0000007f73080fb0  [stack:6968]
         ........  ........
    #01  0000007f73080f80  0000007f73080fb0  [stack:6968]
         0000007f73080f88  0000007f81cea84c  /system/lib64/libc.so (raise+32)
         0000007f73080f90  0000007f73081018  [stack:6968]
         0000007f73080f98  0000007f73081020  [stack:6968]
         0000007f73080fa0  0000007fda87e5e8  [stack]
         0000007f73080fa8  0000000000000091
    #02  0000007f73080fb0  0000007f73080fe0  [stack:6968]
         0000007f73080fb8  0000007f81ce4fec  /system/lib64/libc.so (abort+64)
         0000007f73080fc0  0000007f73081018  [stack:6968]
         0000007f73080fc8  0000007f81870f3c  /system/vendor/lib64/libCB.so (cb_enqueue_write_buffer+284)
         0000007f73080fd0  ffffffffffffffdf
         0000007f73080fd8  0000000000000006
    #03  0000007f73080fe0  0000007f73081040  [stack:6968]
         0000007f73080fe8  0000007f81ce8434  /system/lib64/libc.so (__fortify_chk_fail)
         0000007f73080ff0  0000007fda87e838  [stack]
         0000007f73080ff8  0000007f7d3e067f  anon_inode:dmabuf
         0000007f73081000  0000007f78a096e0  [anon:libc_malloc]
         0000007f73081008  0000000000000001
         0000007f73081010  0000007f78996200  [anon:libc_malloc]
         0000007f73081018  ffffffffffffffdf
         0000007f73081020  0000000000000091
         0000007f73081028  0000007fda87e5f8  [stack]
         0000007f73081030  0000005562bd7000  /data/openpilot/selfdrive/camerad/camerad
         0000007f73081038  7c37eaf726574e1d
    #04  0000007f73081040  0000007f73081150  [stack:6968]
         0000007f73081048  0000007f81ce50c4  /system/lib64/libc.so (longjmperror)
         0000007f73081050  0000007f73081150  [stack:6968]
         0000007f73081058  0000007f73081150  [stack:6968]
         0000007f73081060  0000007f73081110  [stack:6968]
         0000007f73081068  ffffff80ffffffc8
         0000007f73081070  0000007f73081150  [stack:6968]
         0000007f73081078  0000007f73081150  [stack:6968]
         0000007f73081080  0000007f73081110  [stack:6968]
         0000007f73081088  ffffff80ffffffc8
         0000007f73081090  0000000000000000
         0000007f73081098  0000000000000000
         0000007f730810a0  72702e6775626564
         0000007f730810a8  6f697461746f7265
         0000007f730810b0  6c62617369642e6e
         0000007f730810b8  0000000000000065
         ........  ........
    #05  0000007f73081150  0000007f73081380  [stack:6968]
         0000007f73081158  0000005562b996c0  /data/openpilot/selfdrive/camerad/camerad (_Z20camera_process_frameP16MultiCameraStateP11CameraStatei+2400)
    #06  0000007f73081160  0000000000000000
         0000007f73081168  0000007f81d2d000  /system/lib64/libc.so (strpbrk+32)
         0000007f73081170  0000000000000154
         0000007f73081178  0000000000000004
         0000007f73081180  0000000000000000
         0000007f73081188  0000007f73285a00  [anon:libc_malloc]
         0000007f73081190  0000007fda87eef8  [stack]
         0000007f73081198  0000007fda87f7d8  [stack]
         0000007f730811a0  0000007f7edd1700  [anon:libc_malloc]
         0000007f730811a8  0000000000000000
         0000007f730811b0  0000005562bd766c  /data/openpilot/selfdrive/camerad/camerad
         0000007f730811b8  0000007f81d2d000  /system/lib64/libc.so (strpbrk+32)
         0000007f730811c0  0000007fda87e4a0  [stack]
         0000007f730811c8  0000007fda87f7d8  [stack]
         0000007f730811d0  0000007f730811e0  [stack:6968]
         0000007f730811d8  0000007f8186e6dc  /system/vendor/lib64/libCB.so (cl_arraylist_destroy+28)
         ........  ........
    #07  0000007f73081390  0000007f78957150  [anon:libc_malloc]
         0000007f73081398  0000000000000000
         0000007f730813a0  000000000000000b
         0000007f730813a8  00000000000ff000
         0000007f730813b0  0000007f81e770d8
         0000007f730813b8  0000007f81d9f000  /system/lib64/libc.so
         0000007f730813c0  0000007f730814c0  [stack:6968]
         0000007f730813c8  0000007f78957150  [anon:libc_malloc]
         0000007f730813d0  0000007f73081400  [stack:6968]
         0000007f730813d8  0000005562b95008  /data/openpilot/selfdrive/camerad/camerad (_ZNSt6__ndk114__thread_proxyINS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEPFPvP16MultiCameraStatePKcP11CameraStatePFvS9_SD_iEES9_SB_SD_SF_EEEEES7_S7_+56)
    #08  0000007f730813e0  0000007f81d9f000  /system/lib64/libc.so
         0000007f730813e8  0000007f730814c0  [stack:6968]
         0000007f730813f0  0000007f730814c0  [stack:6968]
         0000007f730813f8  0000007f73081440  [stack:6968]
         0000007f73081400  0000007f73081410  [stack:6968]
         0000007f73081408  0000007f81d2dd58  /system/lib64/libc.so (_ZL15__pthread_startPv+56)
    #09  0000007f73081410  0000007f73081430  [stack:6968]
         0000007f73081418  0000007f81ce5b98  /system/lib64/libc.so (__start_thread+20)
         0000007f73081420  0000007f81e77018
         0000007f73081428  0000000000001b02
    #10  0000007f73081430  0000000000000000
         0000007f73081438  0000000000000000
         0000007f73081440  0000007f7327f440
         0000007f73081448  0000007f72f82440
         0000007f'
```

**Comment by deanlee** [comment]
maybe related to this : https://github.com/commaai/openpilot/pull/19555

**Comment by pd0wm** [comment]
Probably fixed in 0.8.2, need to check the results after it's been in release2 for a bit.
